src/main/java/org/opensearch/knn/plugin/script/KNNScoringSpaceUtil.java

```diff
@@ -137,4 +146,43 @@ public static float getVectorMagnitudeSquared(float[] inputVector) {
         }
         return normInputVector;
     }
+
+    /**
+     * Get the expected dimensions from a specified knn vector field type.
+     *
+     * If the field is model-based, get dimensions from model metadata.
+     * @param knnVectorFieldType knn vector field type
+     * @return expected dimensions
+     */
+    public static int getExpectedDimensions(KNNVectorFieldMapper.KNNVectorFieldType knnVectorFieldType) {
+        int expectedDimensions = knnVectorFieldType.getDimension();
```

Comment:
nit: you can make if check as a method and name it something like `isModelBasedIndex()`, this way the flow of getExpectedDimensions is easier to read